### ISSUE TITLE
Index Tables are not working as expected

### ISSUE THREAD

**Issue by farooki** [ownego/polaris-vue#395]
Index tables are not loading properly. I'm getting double headers and on selecting all rows tick mark icon is not appearing. any idea how can I fix it?

![image](https://github.com/user-attachments/assets/aec3ee58-c713-457f-a24b-0049fbc6083d)
![image](https://github.com/user-attachments/assets/63bc1914-2411-457c-8b0b-39f9caf14287)

Also its documentation is not in vue which is causing a lot of confusion.

**Comment by juzser** [comment]
Could you provide your code here then I can check? 

> Also its documentation is not in vue which is causing a lot of confusion.

I don't understand what you mean for this

**Comment by farooki** [comment]
> Could you provide your code here then I can check?
> 
> > Also its documentation is not in vue which is causing a lot of confusion.
> 
> I don't understand what you mean for this. Are you checking this [docs](https://ownego.github.io/polaris-vue/components/IndexTable)?

Hey @juzser , thanks for your reply. I have added vue component below for the reference. Please check 

```vue
import { useIndexResourceState } from '@ownego/polaris-vue';
import { h, resolveComponent } from 'vue';

export default {
    name: 'Table',
    data() {
        return {
            orders: [
                {
                    id: '1020',
                    order: '#1020',
                    date: 'Jul 20 at 4:34pm',
                    customer: 'Jaydon Stanton',
                    total: '$969.44',
                    paymentStatus: () => h(resolveComponent('Badge'), { progress: 'complete' }, () => 'Paid'),
                    fulfillmentStatus: () => h(resolveComponent('Badge'), { progress: 'incomplete' }, () => 'Unfulfilled'),
                },
                {
                    id: '1019',
                    order: '#1019',
                    date: 'Jul 20 at 3:46pm',
                    customer: 'Ruben Westerfelt',
                    total: '$701.19',
                    paymentStatus: () => h(resolveComponent('Badge'), { progress: 'partiallyComplete' }, () => 'Partially paid'),
                    fulfillmentStatus: () => h(resolveComponent('Badge'), { progress: 'incomplete' }, () => 'Unfulfilled'),
                },
                {
                    id: '1018',
                    order: '#1018',
                    date: 'Jul 20 at 3.44pm',
                    customer: 'Leo Carder',
                    total: '$798.24',
                    paymentStatus: () => h(resolveComponent('Badge'), { progress: 'complete' }, () => 'Paid'),
                    fulfillmentStatus: () => h(resolveComponent('Badge'), { progress: 'incomplete' }, () => 'Unfulfilled'),
                },
            ],
            resourceName: {
                singular: 'order',
                plural: 'orders',
            },
            selectedResources: [],
            allResourcesSelected: false,
            handleSelectionChange: null,
        };
    },
    methods: {

    },
    mounted() {
        const { selectedResources, allResourcesSelected, handleSelectionChange } = useIndexResourceState(this.orders);
        this.selectedResources = selectedResources;
        this.allResourcesSelected = allResourcesSelected;
        this.handleSelectionChange = handleSelectionChange;
    },
    template: `
    <div class="p-4">
      <LegacyCard>
            <IndexTable
                :resourceName="resourceName"
                :itemCount="orders.length"
                :selectedItemsCount="allResourcesSelected ? 'All' : selectedResources.length"
                @selection-change="handleSelectionChange"
                :headings="[
                  { title: 'Order' },
                  { title: 'Date' },
                  { title: 'Customer' },
                  { title: 'Total', alignment: 'end' },
                  { title: 'Payment status' },
                  { title: 'Fulfillment status' },
                ]"
            >
                <IndexTableRow
                    v-for="{ id, order, date, customer, total, paymentStatus, fulfillmentStatus }, index in orders"
                    :id="id"
                    :key="id"
                    :position="index"
                    :selected="selectedResources.includes(id)"
                >
                    <IndexTableCell>
                        <Text variant="bodyMd" fontWeight="bold" as="span">{{ order }}</Text>
                    </IndexTableCell>
                    <IndexTableCell>{{ date }}</IndexTableCell>
                    <IndexTableCell>{{ customer }}</IndexTableCell>
                    <IndexTableCell>
                        <Text as="span" alignment="end" numeric>{{ total }}</Text>
                    </IndexTableCell>
                    <IndexTableCell>
                        <component :is="paymentStatus"></component>
                    </IndexTableCell>
                    <IndexTableCell>
                        <component :is="fulfillmentStatus"></component>
                    </IndexTableCell>
                </IndexTableRow>
            </IndexTable>
      </LegacyCard>
    </div>
  `
};

```

**Comment by juzser** [comment]
Could you try to change the `mounted` hook to `setup` hook? 
I doubt the reactivity variables (such as `selectedResources`) must be defined before template rendered.

Otherwise, all looks good.

**Comment by farooki** [comment]
> Could you try to change the `mounted` hook to `setup` hook? I doubt the reactivity variables (such as `selectedResources`) must be defined before template rendered.
> 
> Otherwise, all looks good.

Hi @juzser , I also tried that, but the results remain the same. anything else that I can try? or if you have any support member who can jump into a quick call and look, maybe he/she can identify the issue?

**Comment by HQCuong** [comment]
Hi @farooki, this issue fixed in version [2.1.10](https://github.com/ownego/polaris-vue/releases/tag/v2.1.10)
